src/Kmd.Logic.Gateway.Automation/PreValidation/ApisPreValidation.cs

```diff
@@ -22,8 +30,32 @@ public IEnumerable<GatewayAutomationResult> ValidateAsync(PublishFileModel publi
                     this.ValidationResults.Add(new GatewayAutomationResult { IsError = true, ResultCode = ResultCode.ValidationFailed, Message = $"Duplicate api names exist" });
                 }
 
+                using var client = this.gatewayClientFactory.CreateClient();
+                var products = client.GetAllProducts(this.options.SubscriptionId, this.options.ProviderId);
+                var nameofproducts = products.Select(x => x.Name).ToList();
+                var consolidatedProductNames = nameofproducts.Union(publishFileModel.Products.Select(y => y.Name).ToList());
+
                 foreach (var api in publishFileModel.Apis)
                 {
+                    if (api.ApiVersions == null)
+                    {
+                        continue;
```

Comment:
API should have version error has to come here

Reply:
Added